include/ibamr/LevelSetUtilities.h

```diff
@@ -0,0 +1,428 @@
+                                       bool skip_synchronize_new_state_data,
+                                       int num_cycles,
+                                       void* ctx);
+
+    /*!
+     * \brief Compute the value of the Lagrange multiplier and use that to adjust the flow level set when
+     * there is a solid phase in the domain.
+     */
+    static void fixMassLoss3PhaseFlows(double current_time,
+                                       double new_time,
+                                       bool skip_synchronize_new_state_data,
+                                       int num_cycles,
+                                       void* ctx);
+
+    /*!
+     * \return Integrals of \f$ 1- H(\phi)\f$, \f$ H(\phi)\f$, and \f$ \delta(\phi) \f$  over
+     * the entire domain.
+     *
+     * Here, \f$ H(\phi) \f$ is the Heaviside function demarcating liquid and gas domains.
+     *
+     * \f$ \phi \f$ is taken to be positive in the liquid domain and negative in the gas domain.
+     *
+     * Physically, these three integrals represent volume of the gas region, liquid region, and
+     * surface area of the interface, respectively.
+     */
+    static std::vector<double> computeHeavisideIntegrals2PhaseFlows(LevelSetContainer* lsc);
+
+    /*!
+     * \return Integrals of \f$ [1- H(\phi)] H(\Psi) \f$, \f$ H(\phi) H(\Psi)\f$, \f$ 1 - H(\Psi)\f$,
+     * and \f$ \delta(\phi)H(\Psi)\f$ over the entire domain.
+     *
+     * Here, \f$ H(\phi) \f$ is the Heaviside function demarcating liquid and gas domains,  and \f$ H(\Psi) \f$ is
+     * the Heaviside function demarcating solid and fluid (fluid = liquid and gas) domains.
+     *
+     * Physically, these four integrals represents volume of the gas region, liquid region, solid region, and
+     * surface area of the fluid interface, respectively.
+     *
+     * \f$ \phi \f$ is taken to be positive in the liquid domain and negative in the gas domain.
+     *
+     * \f$ \Psi \f$ is taken to be positive outside the solid and negative inside the solid.
+     */
+    static std::vector<double> computeHeavisideIntegrals3PhaseFlows(LevelSetContainer* lsc);
+
+    /*!
+     * \return Integral of inflow \f$ -\vec{u} \cdot \vec{n}\f$ at a physical boundary.
+     *
+     * \param location_idx of the boundary at which the integral is evaluated
+     */
+    static double computeNetInflowPhysicalBoundary(SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                                   int u_idx,
+                                                   int bdry_loc_idx);
+
+    /*!
+     * \brief Class SetLSProperties is a utility class which sets (or resets after reinitialization)
+     * level set values on the patch hierarchy.
+     */
+    class SetLSProperties
+    {
+    public:
+        /*!
+         * The only constructor of this class.
+         */
+        SetLSProperties(const std::string& object_name, SAMRAI::tbox::Pointer<IBAMR::LSInitStrategy> ls_ops)
```

Comment:
In cases where there are multiple level sets, do you have a single `LSInitStrategy`? Should there be a constructor with `std::vector<Pointer<LSInitStrategy>>`?